### PR TITLE
Implemented changing comment to a ReplicatedMergeTree table

### LINKED ISSUE #36377: ALTER TABLE MODIFY COMMENT does not work on ReplicatedMergeTree
We use Clickhouse version 22.3.3.44, and met with with some wrong behaviour - 
queries "ALTER TABLE ... MODIFY COMMENT ... " does not work on ReplicatedMergeTree engine.

Bug reproduce script:
```
CREATE TABLE sandbox.test__calendar
(
    `dttm` DateTime COMMENT 'date and time',
    `timestamp` Int64 COMMENT 'event timestamp',
    `year_num` Int32 COMMENT 'year number'
)
ENGINE = ReplicatedMergeTree() PARTITION BY year_num ORDER BY timestamp
COMMENT 'Comment text for test table';

SELECT comment FROM system.tables WHERE name like 'test__calendar'; -- 1 row selected with correct comment text

ALTER TABLE sandbox.test__calendar MODIFY COMMENT 'Some new more detailed text of comment';

SELECT comment FROM system.tables WHERE name like 'test__calendar'; -- 1 row selected, but with OLD(!) comment text
```

Maybe it will be helpfull, ALTER TABLE query returns such result:
```
host            |status|error|num_hosts_remaining|num_hosts_active|
----------------+------+----+-----------------------+------------------+
10.7.179.41     |     0|    |                      1|                 0|
ak1-st-data-01  |     0|    |                      0|                 0|
```

### PR DESCRIPTION
**Pull request by Enmk**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Implemented changing comment to a ReplicatedMergeTree table

Closes: #36377

### CODE REVIEW

File: src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp [hunk 1]
```diff
@@ -155,8 +159,18 @@ void ReplicatedMergeTreeTableMetadata::read(ReadBuffer & in)
     else
         index_granularity_bytes = 0;
 
-    if (checkString("constraints: ", in))
-        in >> constraints >> "\n";
+    String verb;
```
Comment (alesapin) [inline src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp:162]: what does `verb` mean here?

File: tests/queries/0_stateless/02302_ReplicatedMergeTree_comment.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+-- Validate that setting/removing and getting comments on ReplicatedMergeTree works
```
Comment (vdimir) [inline tests/queries/0_stateless/02302_ReplicatedMergeTree_comment.sql:1]:
```suggestion
-- Tags: zookeeper
-- Validate that setting/removing and getting comments on ReplicatedMergeTree works
```

**Review by vdimir — APPROVED** [review]
LGTM. Maybe it would be also good to add to integration test (maybe to test_distributed_ddl or new one) to check that comment changed on all nodes.

File: src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp [hunk 3]
```diff
@@ -155,8 +159,18 @@ void ReplicatedMergeTreeTableMetadata::read(ReadBuffer & in)
...
+    readStringUntilWhitespace(verb, in);
+
+    if (verb == "constraints:")
+    {
+        in >> " " >> constraints >> "\n";
+
+        readStringUntilWhitespace(verb, in);
```
Comment (davenger) [inline src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp:169]: Isn't it possible for "constraints" to contain a whitespace?
I suggest to add test cases with constraints.

**Review by alesapin — COMMENTED** [review]
Do we really need comments to be replicated and stored in ZooKeeper? Maybe it's enough just to change local metadata? It will allow having different comments on different replicas. We do it for settings for example.

File: src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp (hunk 1 above)
Comment (Enmk) [reply src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp:162]: Maybe 'action' would be a better choice...

File: tests/queries/0_stateless/02302_ReplicatedMergeTree_comment.sql (hunk 2 above)
Comment (Enmk) [reply tests/queries/0_stateless/02302_ReplicatedMergeTree_comment.sql:1]: Not needed anymore

**Review by alesapin — COMMENTED** [review]
Failures unrelated.

**Comment by tavplubix** [comment]
This PR breaks test `00992_system_parts_race_condition_zookeeper_long` with `Replicated` database:
https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZIb3VyKGNoZWNrX3N0YXJ0X3RpbWUpIGFzIGQsCmNvdW50KCksIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpCmZyb20gY2hlY2tzIHdoZXJlICcyMDIyLTAzLTAxJyA8PSBjaGVja19zdGFydF90aW1lIGFuZCB0ZXN0X25hbWUgbGlrZSAnMDA5OTJfc3lzdGVtX3BhcnRzX3JhY2VfY29uZGl0aW9uX3pvb2tlZXBlcl9sb25nJScgYW5kIGNoZWNrX25hbWUgaWxpa2UgJyVyZXBsJScgYW5kIHRlc3Rfc3RhdHVzIGluICgnRkFJTCcsICdGTEFLWScsICdFUlJPUicpIGdyb3VwIGJ5IGQgb3JkZXIgYnkgZCBkZXNj

**Comment by alexey-milovidov** [comment]
@Enmk it does not work at all.

**Comment by alexey-milovidov** [comment]
It is ok to have COMMENT not replicated, similarly to, e.g. storage policy.

**Pull request merged**